launch_ros/launch_ros/actions/node.py

```diff
@@ -432,3 +481,86 @@ def expanded_node_namespace(self):
     def expanded_remapping_rules(self):
         """Getter for expanded_remappings."""
         return self.__expanded_remappings
+
+    def _get_extensions(self):
```

Comment:
@gbiggs nit: consider making this a free function ~or `staticmethod`~ (better not, too much support logic inside the `Node` class) that returns the list of extensions.

Reply:
This function accesses the node's logger to output an error message that is specific to the function's internal implementation. I could refactor that out into returning an optional error message, but I'm wary of going to far for something that's meant to be temporary. How strongly do you feel about this?

Reply:
I've made the two functions free in 0d9cb67.

Reply:
> Until we get around to replace it, it is permanent

Fair point! :)